### ISSUE TITLE
problem with wrong device interface getting route assignment

### ISSUE THREAD

**Issue by lthurlow** [lthurlow/scaling-computing-machine#3]
MSG: [
{"Use": "0", "Iface": "eth1", "Metric": "1", "Destination": "10.0.0.0", "Genmask": "255.255.255.0", "Flags": "U", "TTL": "2015194112431241960", "Ref": "0", "Gateway": "10.0.1.2"},
 {"Use": "0", "Iface": "eth2", "Metric": "1", "Destination": "10.0.1.0", "Genmask": "255.255.255.0", "Flags": "U", "TTL": "2015194112431241960", "Ref": "0", "Gateway": "10.0.1.2"}, 
{"Use": "0", "Iface": "eth3", "Metric": "1", "Destination": "10.0.2.0", "Genmask": "255.255.255.0", "Flags": "U", "TTL": "2015194112431241960", "Ref": "0", "Gateway": "10.0.1.2"}]

Eth3 does not exist on an2, seems to not get modified correctly when receiving from an3

**Comment by lthurlow** [comment]
eth2 is still clobbering eth1, at which point the route tables get dropped and everything goes to hell.

AN2: Interface eth1

```
[10.0.0.1.38667.Thread-1:51 - MainThread             <module>]       INFO Starting RIP server on [eth1]
[10.0.0.1.38667.Thread-1:53 - MainThread             <module>]      DEBUG 10.0.0.1.38667.Thread-1 Thread: RIP Server
[10.0.0.1.38667.Thread-1:54 - MainThread             <module>]      DEBUG PID: 1473
[rip.py:228 - Thread-1           rip_server]      DEBUG RIP SERVER:
[rip.py:229 - Thread-1           rip_server]      DEBUG PID: 1473
[rip.py:230 - Thread-1           rip_server]      DEBUG DEVICES: eth1
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[rip.py:40 - Thread-1           write_n_fi]      DEBUG      WRITE_N_FI
[rip.py:43 - Thread-1           write_n_fi]      DEBUG          writing: {'Use': '0', 'Iface': 'eth1', 'Metric': '0', 'Destination': '10.0.0.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408638269', 'Ref': '0', 'Gateway': '0.0.0.0'}
[rip.py:43 - Thread-1           write_n_fi]      DEBUG          writing: {'Use': '0', 'Iface': 'eth2', 'Metric': '0', 'Destination': '10.0.1.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408642127', 'Ref': '0', 'Gateway': '0.0.0.0'}
```

AN2: Interface eth2

```
[<string>:67 - Thread-1              sandbox]      DEBUG    process started: 1478
[10.0.1.2.56859.Thread-1:35 - MainThread             <module>]      DEBUG Checking for RIP file: .route_ripeth2
[10.0.1.2.56859.Thread-1:51 - MainThread             <module>]       INFO Starting RIP server on [eth2]
[10.0.1.2.56859.Thread-1:53 - MainThread             <module>]      DEBUG 10.0.1.2.56859.Thread-1 Thread: RIP Server
[10.0.1.2.56859.Thread-1:54 - MainThread             <module>]      DEBUG PID: 1478
[rip.py:228 - Thread-1           rip_server]      DEBUG RIP SERVER:
[rip.py:229 - Thread-1           rip_server]      DEBUG PID: 1478
[rip.py:230 - Thread-1           rip_server]      DEBUG DEVICES: eth2
[rip.py:48 - Thread-1            read_n_fi]      DEBUG      READ_N_FI
[rip.py:53 - Thread-1            read_n_fi]      DEBUG          loaded: {'Use': '0', 'Iface': 'eth1', 'Metric': '0', 'Destination': '10.0.0.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408638269', 'Ref': '0', 'Gateway': '0.0.0.0'}
[rip.py:53 - Thread-1            read_n_fi]      DEBUG          loaded: {'Use': '0', 'Iface': 'eth2', 'Metric': '0', 'Destination': '10.0.1.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408642127', 'Ref': '0', 'Gateway': '0.0.0.0'}
[rip.py:56 - Thread-1            read_n_fi]       INFO [{'Use': '0', 'Iface': 'eth1', 'Metric': '0', 'Destination': '10.0.0.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408638269', 'Ref': '0', 'Gateway': '0.0.0.0'}, {'Use': '0', 'Iface': 'eth2', 'Metric': '0', 'Destination': '10.0.1.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003408642127', 'Ref': '0', 'Gateway': '0.0.0.0'}]
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[anhost.py:115 - Thread-1            set_route]      ERROR KeyError with adding TTL
[rip.py:40 - Thread-1           write_n_fi]      DEBUG      WRITE_N_FI
[rip.py:43 - Thread-1           write_n_fi]      DEBUG          writing: {'Use': '0', 'Iface': 'eth0', 'Metric': '0', 'Destination': '192.168.122.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003409625080', 'Ref': '0', 'Gateway': '0.0.0.0'}
[rip.py:43 - Thread-1           write_n_fi]      DEBUG          writing: {'Use': '0', 'Iface': 'eth0', 'Metric': '0', 'Destination': '192.168.122.0', 'Genmask': '255.255.255.0', 'Flags': 'U', 'TTL': '2015196003409629078', 'Ref': '0', 'Gateway': '0.0.0.0'}
```

**Comment by lthurlow** [comment]
from build: eb41cc1ed712524fe33b6d80f3631efbecb19fae

AN1 and AN2, eth1 had an issue communicating.  AN1 would send broadcast, but did not receive updates. So AN1's route was distributed, but AN1 did not get any other routes.

Timeouts still work.

**Comment by lthurlow** [comment]
Looks like what happens is that AN2-2 is overwriting AN2-1's routes still, and this causes AN2 to not broadcast to AN1.

**Comment by lthurlow** [comment]
fixed in: c66912e7686acc106537ecb0647b9aece0a55015

when the second interface boots it shouldnt tamper with the tables.
